### PR TITLE
Add worker scheduler tests

### LINKED ISSUE #1764: Add tests for the worker scheduler
Please add tests for the worker scheduler.

### PR DESCRIPTION
**Pull request by igorschoester**
## Summary

This PR can be summarized in the following changelog entry:

* N/A

## Relevant technical choices:

* Fixed stop polling in the scheduler by:
  - `tick` saves the setTimeout handle to `_pollHandle`.
  - `stopPolling` sets `_started` to `false`.
* Added tests for worker scheduler and worker scheduler task.

## Test instructions

This PR can be tested by following these steps:

* Check out the branch: `git checkout 1764-add-worker-scheduler-tests`.
* Install packages: `yarn`.
* Run the tests: `yarn test`.
* Verify the tests succeed.
* Open the coverage file for the scheduler: `/coverage/lcov-report/src/worker/scheduler/index.html`.
* Verify that the coverage is 100%!

Fixes #1764

### CODE REVIEW

File: spec/worker/scheduler/SchedulerSpec.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+			expect( scheduler._tasks.analyze.length ).toBe( 0 );
+
+			scheduler.schedule( { id: 0, execute: () => {}, done: () => {}, data: { test: true }, type: "analyze" } );
+
+			expect( scheduler._tasks.analyze.length ).toBe( 1 );
+			expect( scheduler._tasks.analyze[ 0 ].id ).toBe( 0 );
+			expect( scheduler._tasks.analyze[ 0 ].data ).toEqual( { test: true } );
+		} );
+
+		test( "getNextTask to prioritize by type", () => {
```
Comment (nataliashitova) [inline spec/worker/scheduler/SchedulerSpec.js]:
```suggestion
		test( "getNextTask prioritizes tasks by type", () => {
```

File: src/worker/scheduler/Scheduler.js [hunk 2]
```diff
@@ -4,11 +4,12 @@ import { merge } from "lodash-es";
 // Internal dependencies.
 import Task from "./Task";
 
+
 const DEFAULT_CONFIGURATION = {
 	pollTime: 50,
 };
 
-class Scheduler {
+export default class Scheduler {
```
Comment (nataliashitova) [inline src/worker/scheduler/Scheduler.js]: Eslint complains about the missing JSdoc comment here.
Please don't forget to adjust the eslint warning limits after fixing! 😄

Comment (igorschoester) [reply src/worker/scheduler/Scheduler.js]: I'm the right person to 'boy scout' this I suppose... done!

**Comment by nataliashitova** [comment]
Acceptance 👍

**Pull request merged**